extension/core_functions/scalar/list/array_slice.cpp

```diff
@@ -59,18 +59,25 @@ static idx_t CalculateSliceLength(idx_t begin, idx_t end, INDEX_TYPE step, bool
 }
 
 template <typename INPUT_TYPE, typename INDEX_TYPE>
-INDEX_TYPE ValueLength(const INPUT_TYPE &value) {
+INDEX_TYPE ValueLength(const INPUT_TYPE &value, const LogicalTypeId &) {
 	return 0;
 }
 
 template <>
-int64_t ValueLength(const list_entry_t &value) {
+int64_t ValueLength(const list_entry_t &value, const LogicalTypeId &) {
 	return UnsafeNumericCast<int64_t>(value.length);
 }
 
 template <>
-int64_t ValueLength(const string_t &value) {
-	return LengthFun::Length<string_t, int64_t>(value);
+int64_t ValueLength(const string_t &value, const LogicalTypeId &type) {
+	switch (type) {
+	case LogicalTypeId::BLOB:
```

Comment:
Perhaps instead of passing in the `LogicalType` here, we can put these functions in a struct and use that as a template parameter? e.g.:

```cpp
// ...

struct BlobSliceOperations {
    static int64_t ValueLength(const string_t &value) {
        return value.GetSize();
    }
    // ...
};


struct StringSliceOperations {
    static int64_t ValueLength(const string_t &value) {
        return LengthFun::Length<string_t, int64_t>(value);
    }
    // ...
};

```

Then pass it as a template parameter `class OP`, and use the functions using `OP::ValueLength`